pkg/modelzooserver/input_check.go

```diff
@@ -23,7 +23,7 @@ func checkName(name string) error {
 	if len(name) < 6 {
 		return fmt.Errorf("model name should have at least 6 characters")
 	}
-	match, err := regexp.MatchString(`^[a-zA-Z0-9_-]{6,256}$`, name)
+	match, err := regexp.MatchString(`^[a-zA-Z0-9_\-]{6,256}$`, name)
```

Comment:
when '-' is the first or the last char in '[]', we do not need to escape it

Reply:
Good point. Will refine in following pull requests.